### ISSUE TITLE
Instaling poetry.extras

### ISSUE THREAD

**Issue by phschiele** [robertmartin8/PyPortfolioOpt#264]
**Is your feature request related to a problem? Please describe.**
I was setting up PyPortfolioOpt on a new machine using poetry. 
After cloning, I ran `poetry install`, which installed all packages from the `dependencies` and `dev-dependencies` section in the [pyproject.toml](https://github.com/robertmartin8/PyPortfolioOpt/blob/master/pyproject.toml) file. To run the tests, I also need the packages from the `extras` section. However, I was not sure how to install them using `poetry`.

**Describe the solution you'd like**
Changing the file to something like the example below would allow for an easy installation using `poetry install -E optionals`.
```
[tool.poetry.dependencies]
python = "^3.5"
numpy = "^1.12"
...
scikit-learn = { version = "^0.19", optional = true }
matplotlib = { version = "^3.2.0", optional = true }


[tool.poetry.dev-dependencies]
...

[tool.poetry.extras]
optionals = ["scikit-learn", "matplotlib"]
```

**Additional context**
- I ran into some version conflicts when adding the specified matplotlib version to the dependency tree, perhaps the version needs to be bumped
- I wanted to suggest using GitHub actions to automatically run pytest, flake8 and black on every PR. Its [free for public repos](https://docs.github.com/en/github/setting-up-and-managing-billing-and-payments-on-github/about-billing-for-github-actions) and IMO really helpful for everyone contributing to the project.

Would be great to hear your thoughts on the poetry and GitHub Actions aspects.

**Comment by robertmartin8** [comment]
@phschiele Thanks for raising this!

Will definitely fix the poetry thing – when I first created my `pyproject.toml`, I couldn't figure out the best way of having optional dependencies, but your suggestion makes sense.

Noted re the matplotlib version. Will try to fix it with v1.3.0.

I definitely need to read more about GitHub actions, all the cool kids seem to be using it! But I wonder which aspects of the current workflow it improves?

For PRs from contributors, I normally merge them into a new dev branch which I work on locally (so my editor applies `black`). When I PR into master, Travis CI runs `pytest` (results are [here](https://travis-ci.org/github/robertmartin8/PyPortfolioOpt/branches)).

**Comment by phschiele** [comment]
I think the advantage of having the extra packages in `dependencies` section (even though optional), is that poetry will already consider the version constraints, such that the versions specified are guaranteed to be compatible with the other dependencies. 

As for GitHub actions, IMO the main advantage is that everything is contained in one service. I am not too familiar with Travis, but another nice thing about GH actions is that there exists a vast marketplace with prebuilt actions. If you think it is generally interesting, I'd be happy to create an example in my fork of the repo. I guess seeing it in action is better than explaining it :)

**Comment by phschiele** [comment]
@robertmartin8 Check out [this PR](https://github.com/phschiele/PyPortfolioOpt/pull/) in my fork. 
It's just an example, though. It is straightforward to extend it with dependency caching, black, coverage, typing, and [much more](https://github.com/marketplace?type=actions).

**Comment by robertmartin8** [comment]
@phschiele 

Completely agreed re `poetry`, will push that into v1.3.0. I've been trying to figure out the issues you mentioned re the dependency resolution. A big problem I'm facing is PyPortfolioOpt's support for python 3.5. Matplotlib no longer supports this; neither does black. The problem is that `zipline` requires python 3.5... I thought this was important in the past, but perhaps with the shutdown of quantopian, less so now. I will likely drop 3.5 support in v1.3.0.

GitHub actions sound very interesting. Actually, it just so happens that Travis CI is migrating from `.org` to `.com` in the next few weeks, meaning that I would essentially have to reconfigure. This means that now is probably as good at time as any for me to migrate to GH actions.

I'm a bit pressed for time, so will just be looking to implement the bare basics right now: automatically running `pytest` for python 3.6 -> 3.8 for every PR. Do you have any links on how to get started with it?

In future, we can add other stuff like `black` and coverage.

**Comment by phschiele** [comment]
@robertmartin8 I see, I also ran into the 3.5 issue with `zipline` that you described before.

Sounds like a good  time to switch to GH Actions then.
Please take another look at [my PR](https://github.com/phschiele/PyPortfolioOpt/pull/2). It pretty much adds everything you ask for and even implements caching (the dependency installation took longer than the tests themself). I have also added some comments about test sensitivities in that PR.

Check out the [GitHub Docs](https://docs.github.com/en/actions/guides/building-and-testing-python) about running pytest w/ GH actions for further reference.

**Comment by robertmartin8** [comment]
@phschiele Cheers, will check out the PR (the first link was dead).

**Comment by robertmartin8** [comment]
Ok, I've copy-pasted the action into `master`. It's failing now because the `poetry` file is still in v1.3.0, but once that gets merged, it should run! 

Thanks for suggesting GH actions, it honestly seems like it's gonna save me a huge amount of time in the long run!

**Comment by phschiele** [comment]
Cheers, glad you like it!
Two final (small) comments:
- `.travis.yml` would be obsolete
- You probably want to change your badge in the README to something like 
`![Build Status](http://github.com/robertmartin8/PyPortfolioOpt/workflows/pytest/badge.svg?event=push)`
=> This way, it will only consider the pushes to master for the badge (you don't want a build failing in a PR leading to a "failing" badge in the README.

**Comment by robertmartin8** [comment]
Yup, will clean out all the Travis stuff once I've merged v1.3.0 and verify that all aspects of my previous workflow have been ported over.

**Comment by robertmartin8** [comment]
Fixed with v1.3.0
